docs/DeveloperGuide.md

```diff
@@ -1,34 +1,216 @@
 # Developer Guide
+- [Design](#1-Design)
+- [Implementation](#2-implementation)
+  - [[Proposed] Subject Feature](#21-proposed-subject-feature)
+- [Appendix A: Product Scope](#appendix-a-product-scope)
+- [Appendix B: User Stories](#appendix-b--user-stories)
+- [Appendix C: Non-Functional Requirements](#appendix-c-non-functional-requirements)
+- [Appendix D: Glossary](#appendix-d-glossary)
+- [Appendix E: Instructions for Manual Testing](#appendix-e-instructions-for-manual-testing)
 
-## Design & Implementation
+## 1. Design
+### Architecture
+The Duke class is the main class of the product. It is responsible for:
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
+* At shut down: Shuts down the components and invokes storage method where necessary.	
 
```

Comment:
How do the different components of the app interact with one another? Perhaps you could add an Architecture Diagram to explain the high-level design of the app.